test/IISIntegration.FunctionalTests/Utilities/IISFunctionalTestBase.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Xunit.Abstractions;
+
+namespace IISIntegration.FunctionalTests.Utilities
+{
+    public class IISFunctionalTestBase : FunctionalTestsBase
+    {
+        public IISFunctionalTestBase(ITestOutputHelper output = null) : base(output)
+        {
+        }
+
+        protected override Task<IISDeploymentResult> DeployAsync(DeploymentParameters parameters)
+        {
+            if (parameters.SiteName == null)
+            {
+                parameters.SiteName = "HttpTestSite";
+            }
+            if (parameters.ServerConfigTemplateContent == null)
+            {
+                parameters.ServerConfigTemplateContent = File.ReadAllText("AppHostConfig/Http.config");
```

Comment:
We have to carry default Http.config that fits almost everyone, and instead of providing a way to replace it completely provide a way to add the XML transformation callback,